lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -289,6 +290,38 @@ impl ShardHolder {
             }
         };
 
+        // Cleanup existing shards if resharding down
+        if is_in_progress && direction == ReshardingDirection::Down {
+            for (&id, shard) in self.shards.iter() {
+                // Skip shards that does not belong to resharding shard key
+                if self.shard_id_to_key_mapping.get(&id) != shard_key.as_ref() {
+                    continue;
+                }
+
+                // Skip target shard
+                if id == shard_id {
+                    continue;
+                }
+
+                // Revert replicas in `Resharding` state back into `Active` state
+                for (peer, state) in shard.peers() {
+                    if state == ReplicaState::Resharding {
+                        shard.set_replica_state(&peer, ReplicaState::Active)?;
+                    }
+                }
+
+                // We only cleanup local shards
+                if !shard.is_local().await {
+                    continue;
+                }
+
+                // Remove any points that might have been transferred from target shard
+                let filter = self.hash_ring_filter(id).expect("hash ring filter");
+                let filter = Filter::new_must(Condition::CustomIdChecker(Arc::new(filter)));
+                shard.cleanup_local_shard(filter).await?;
```

Comment:
Judging by the change I made [here](https://github.com/qdrant/qdrant/pull/5024#discussion_r1762580792), we might need `new_must_not` here.

Reply:
Yeah, I also think that we do. Done.

(And anyway, I'm working on tests for this as part of #5069, so I'll fix/tweak it later if needed.)